train.py

```diff
@@ -1,21 +1,95 @@
+import time
+import torch
+
+
+def train_single_epoch(epoch, model, train_loader, optimizer, criterion, device):
+	"""
+	Train single epoch
+	"""
+	model.train()
+	epoch_start_time = time.time()
+	total_loss=0.
+	for i, batch in enumerate(iter(train_loader)):
+
+		img, target = batch
+		img = img.to(device)
+		target = target.to(device)
+		
+		optimizer.zero_grad()
+		#TODO: introduce for loop to make sentence
+		output, attentions, _ = model(img, target)
+
+		loss = criterion(output, target)
+		loss.backward()
+
+		torch.nn.utils.clip_grad_norm_(model.parameters(), max_norm=0.25)
+
+		optimizer.step()
+
+		total_loss += loss.item()
+
+		log_interval = 100
```

Comment:
log_interval should be defined outside of this function right?

Reply:
It should...though I left it here because we may not change it. I'll take it outside!